src/components/Api.js

```diff
@@ -0,0 +1,114 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+    this._authorization = headers.authorization;
+    this._contentType = headers["Content-Type"];
+  }
+
+  //Check if JSON object ok
+  _checkResponseOk(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      Promise.reject(`Erorr: ${res.status}`)
+    }
+  }
+
+  //Get initial cards from URL
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    }).then(this._checkResponseOk)
+    .catch((err) => {
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.

Please, remove it from `Api` and add it in `index.js`.